numpy/lib/tests/test_io.py

```diff
@@ -1823,6 +1828,123 @@ def test_auto_dtype_largeint(self):
         assert_equal(test['f1'], 17179869184)
         assert_equal(test['f2'], 1024)
 
+
+class TestPathUsage(TestCase):
+    @contextlib.contextmanager
+    def with_path(self, prefix='numpy_test_path_instance_', *args, **kwargs):
+        """
+        Create a temporary directory and a tempfile, and yield the path for the tempfile.
+        
+        args and kwargs are passed to mkstmp.
+        
+        Deletes the temporary directory when with statement finishes.
+        """
+        # Used as in TestSavezLoad.test_closing_zipfile_after_load
+        
+        with tempdir(prefix=prefix) as tmpdir:
+            fd, tmp = mkstemp(dir=tmpdir, *args, **kwargs)
+            os.close(fd)
+            yield Path(tmp)
+    
+    # Test that pathlib.Path can be used
+    @np.testing.dec.skipif(Path is None,
+    "Cannot test without a pathlib.Path instance")
+    def test_loadtxt(self):
+        with self.with_path(prefix='numpy_test_path_loadtxt_', suffix='txt') as path:
+            a = np.array([[1.1, 2], [3, 4]])
+            np.savetxt(path, a)
+            x = np.loadtxt(path)
+            assert_array_equal(x, a)
+
+    @np.testing.dec.skipif(Path is None,
+    "Cannot test without a pathlib.Path instance")
+    def test_save_load(self):
+        # Test that pathlib.Path instances can be used with savez.
+        with self.with_path(prefix='numpy_test_path_save_load_', suffix='.npy') as path:
```

Comment:
Umm, need to keep lines < 80 characters. This looks marginal, although can't tell on screen.